### PR TITLE
display PID of process holding lock

### PR DESCRIPTION
**Pull request by zephaniahong**
Displays PID of process holding lock when trying to run multiple instances of x.py

### CODE REVIEW

File: src/bootstrap/bin/main.rs [hunk 1]
```diff
@@ -20,24 +22,30 @@ fn main() {
...
-                }
+                println!("warning: build directory locked by process {pid}, waiting for lock");
                 t!(build_lock.write())
             }
         };
     }
-    #[cfg(any(not(any(unix, windows)), target_os = "solaris"))]
-    println!("warning: file locking not supported for target, not locking build directory");
+    // #[cfg(any(not(any(unix, windows)), target_os = "solaris"))]
+    // println!("warning: file locking not supported for target, not locking build directory");
```
Comment (jyn514) [inline src/bootstrap/bin/main.rs:48]: why did this change?

File: src/bootstrap/bin/main.rs [hunk 2]
```diff
@@ -20,24 +22,30 @@ fn main() {
...
+            }
             err => {
                 drop(err);
-                if let Some(pid) = get_lock_owner(&path) {
-                    println!("warning: build directory locked by process {pid}, waiting for lock");
-                } else {
-                    println!("warning: build directory locked, waiting for lock");
-                }
+                println!("warning: build directory locked by process {pid}, waiting for lock");
                 t!(build_lock.write())
```
Comment (jyn514) [inline src/bootstrap/bin/main.rs:43]: i think you can fix your problem with a small change here:
```suggestion
                let lock = t!(build_lock.write());
                lock.write(process::id().to_string());
                lock
```

File: src/bootstrap/bin/main.rs (hunk 1 above)
Comment (zephaniahong) [reply src/bootstrap/bin/main.rs:48]: This was an accident. Thanks for the catch!

**Comment by Mark-Simulacrum** [comment]
r=me with description updated to reflect fixed PID writing

**Comment by zephaniahong** [comment]
@Mark-Simulacrum to clarify, do you need me to just edit the description in this PR?

**Comment by Mark-Simulacrum** [comment]
Yes, thanks! @bors r+ rollup

**Pull request merged**